Sources/Basics/AuthorizationProvider.swift

```diff
@@ -53,10 +59,12 @@ private extension URL {
 
 // MARK: - netrc
 
-public struct NetrcAuthorizationProvider: AuthorizationProvider {
+public class NetrcAuthorizationProvider: AuthorizationProvider, AuthorizationWriter {
     // marked internal for testing
     internal let path: AbsolutePath
     private let fileSystem: FileSystem
+    
+    private let cache = ThreadSafeKeyValueStore<String, (user: String, password: String)>()
```

Comment:
This is for adding credentials to memory so `RegistryClient` can read them for the `login` request.